glib-macros/src/lib.rs

```diff
@@ -734,3 +734,74 @@ pub fn variant_derive(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     variant_derive::impl_variant(input)
 }
+
+use quote::quote;
+#[proc_macro_derive(Props, attributes(prop))]
+pub fn derive_props(input: TokenStream) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = input.ident;
+    let properties_match: Vec<syn::Field> = fields_prop(input.data).collect();
+    let match_text = properties_match.iter().map(|field| {
+        let field_name = &field.ident;
+        let field_name_str = field.ident.as_ref().unwrap().to_string();
+        quote!(#field_name_str => self.#field_name.get())
+    });
+    let match_text_set = properties_match.iter().map(|field| {
+        let field_name = &field.ident;
+        let field_name_str = field.ident.as_ref().unwrap().to_string();
+        quote!(#field_name_str => self.#field_name.set(value))
+    });
+    let n_props = properties_match.len();
+    let properties = properties_match.iter().map(|prop| {
+        let ty = &prop.ty;
+        let attr_data: &syn::Attribute =
+            prop.attrs.iter().find(|a| a.path.is_ident("prop")).unwrap();
+        let tokens = attr_data.parse_meta();
+        if let Ok(syn::Meta::List(ls)) = tokens {
+            let nested = ls.nested;
+            quote!(<#ty as glib::Parametrize>::Spec::new(#nested))
+        } else {
+            unreachable!("Invalid attribute params")
+        }
+    });
+    let expanded = quote! {
+        use glib::{Parametrize, ParametrizeMut};
+        impl ObjectImpl for #name {
+            fn properties() -> &'static [glib::ParamSpec] {
+                use once_cell::sync::Lazy;
```

Comment:
Can't you use the re-exported one from glib?

Reply:
Ok right